### PR TITLE
Int and timestamp range operations

### PR DESCRIPTION
**Pull request by ryandillinfelton**
# Overview

This PR is being created to address [ESSNTL-2084](https://issues.redhat.com/browse/ESSNTL-2084) and [ESSNTL-2162](https://issues.redhat.com/browse/ESSNTL-2162).

I think it's reasonable to expect users would like to use range operations on integer and time-stamp fields. Xjoin-search already supports it so we just had to forward the right operations on and adjust our validation. Unfortunately this added some unused variables to the builder functions for the sake of having a uniform interface, but they may need that information down the line so I think it's acceptable.

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: tests/test_xjoin.py [hunk 1]
```diff
@@ -2981,6 +2981,133 @@ def test_generic_filtering_integer(
...
 
 
+def test_generic_filtering_integer_range(
+    mocker,
+    subtests,
+    query_source_xjoin,
+    graphql_query_empty_response,
+    graphql_tag_query_empty_response,
+    graphql_system_profile_sap_system_query_empty_response,
+    patch_xjoin_post,
```
Comment (kruai) [inline tests/test_xjoin.py:2991]: This applies to all these tests, but is this param required? I don't see the function used, and `query_source_xjoin` is already used.

**Review by kruai — APPROVED** [review]
I tried to find a way to reduce the need for extra parameters and return values, but I don't think we can without a big refactor or something. I just have one comment, but this otherwise looks good with thorough tests

**Review by thearifismail — APPROVED** [review]
LGTM
